sdk/tables/azure-data-tables/samples/async_samples/sample_query_tables_async.py

```diff
@@ -38,25 +38,23 @@ async def tables_in_account(self):
         try:
             # [START tsc_list_tables]
             # List all the tables in the service
-            list_tables = table_service.list_tables()
             print("Listing tables:")
-            for table in list_tables:
+            async for table in table_service.list_tables():
                 print("\t{}".format(table.table_name))
             # [END tsc_list_tables]
 
             # [START tsc_query_tables]
             # Query for "table1" in the tables created
             table_name = "mytable1"
             name_filter = "TableName eq '{}'".format(table_name)
-            queried_tables = table_service.query_tables(filter=name_filter, results_per_page=10)
-
             print("Queried_tables")
-            for table in queried_tables:
+            async for table in table_service.query_tables(filter=name_filter, results_per_page=10):
                 print("\t{}".format(table.table_name))
             # [END tsc_query_tables]
 
         finally:
             await self.delete_tables()
+            await table_service.close()
```

Comment:
If you close the parent service client, you wont need to close the table client as well.